lib/lnc.ts

```diff
@@ -137,27 +141,39 @@ export default class LNC {
         // make sure the WASM client binary is downloaded first
         if (!this.isReady) await this.preload();
 
-        global.onLocalPrivCreate = (keyHex: string) => {
-            log.debug('local private key created: ' + keyHex);
-            this.credentials.localKey = keyHex;
-        };
-
-        global.onRemoteKeyReceive = (keyHex: string) => {
-            log.debug('remote key received: ' + keyHex);
-            this.credentials.remoteKey = keyHex;
-        };
+        // create the namespace object in the global scope if it doesn't exist
+        // so that we can assign the WASM callbacks to it
+        if (typeof this.wasm !== 'object') {
+            this.wasm = {};
+        }
 
-        global.onAuthData = (keyHex: string) => {
-            log.debug('auth data received: ' + keyHex);
-        };
+        // assign the WASM callbacks to the namespace object if they haven't
+        // already been assigned by the consuming app
+        if (!this.wasm.onLocalPrivCreate) {
+            this.wasm.onLocalPrivCreate = (keyHex: string) => {
+                log.debug('local private key created: ' + keyHex);
+                this.credentials.localKey = keyHex;
+            };
+        }
+        if (!this.wasm.onRemoteKeyReceive) {
+            this.wasm.onRemoteKeyReceive = (keyHex: string) => {
+                log.debug('remote key received: ' + keyHex);
+                this.credentials.remoteKey = keyHex;
+            };
+        }
+        if (!this.wasm.onAuthData) {
+            this.wasm.onAuthData = (keyHex: string) => {
+                log.debug('auth data received: ' + keyHex);
+            };
+        }
```

Comment:
Yes, we currently don't expose a way for apps to obtain the session auth data (macaroon), which would be useful if the app wanted to decode it and check the caveats. For example, if it wanted to see if the session was for a custodial account. I figured this could be a way to make that possible. Just figured I would throw the checks in there to be cautious.